### PR TITLE
add ci workflow to proto template

### PR DESCRIPTION
**Pull request by shan4993**

### CODE REVIEW

File: t/proto_template/.github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          install_path: '${{ runner.temp }}/.prototool'
+          checksum: '50e53589b95e2435e26de01de8eac6af6ae6980b1cacf1e0ad319b7644f16b3b'
+          cache_key: '${{ runner.os }}_${{ runner.arch }}_prototool_${{ env.PROTOTOOL_VERSION }}'
+          add_to_path: true
+
+      - name: 'Lint proto files'
+        env:
+          PROTOTOOL_PATH: '${{ runner.temp }}/.prototool/prototool/bin/prototool'
+        run: |
+          LINTER_OUTPUT=$(echo $($PROTOTOOL_PATH lint protos))
```
Comment (drevell) [inline t/proto_template/.github/workflows/ci.yml:59]: Is this `echo` unnecessary? I'm probably missing something, but I would have thought that `X=$(echo $(MY_CMD))` is equivalent to just `X=$(MY_CMD)`

File: t/proto_template/spec.yaml [hunk 2]
```diff
@@ -17,22 +17,29 @@ kind: 'Template'
...
   - desc: 'Include some files and directories'
     action: 'include'
     params:
-      paths: ['protos', 'scripts']
+      paths: ['protos', 'scripts', '.github', 'prototool.yaml']
   - desc: 'Replace "github_org_name" and "github_repo_name" with your organization and repo name.'
     action: 'string_replace'
     params:
-      paths: ['protos/v1/person_service/person.proto', 'scripts/generate.sh']
+      paths: ['protos/v1/person_service/person.proto', 'scripts/generate.sh', '.github/workflows/ci.yml', 'prototool.yaml']
```
Comment (drevell) [inline t/proto_template/spec.yaml:38]: If you want, you can simplify by doing `paths: ['.']` to just replace in every file that has been included.

**Review by drevell — CHANGES_REQUESTED** [review]

File: t/proto_template/.github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        with:
+          download_url: 'https://github.com/protocolbuffers/protobuf/releases/download/v${{ env.PROTOBUF_VERSION }}/protoc-${{ env.PROTOBUF_VERSION }}-linux-x86_64.zip'
+          install_path: '${{ runner.temp }}/.protoc'
+          checksum: 'fc793561283d9ea6813fb757ae54f1afea6770afcd930904bdf3e590910420aa'
+          cache_key: '${{ runner.os }}_${{ runner.arch }}_protoc_${{ env.PROTOBUF_VERSION }}'
+          add_to_path: true
+
+      - name: 'Install protoc-gen-go'
+        run: |
+          go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.31
```
Comment (sethvargo) [inline t/proto_template/.github/workflows/ci.yml:44]: Do we want to pin to a specific version of this too?

File: t/proto_template/.github/workflows/ci.yml (hunk 1 above)
Comment (shan4993) [reply t/proto_template/.github/workflows/ci.yml:59]: ack removing

File: t/proto_template/spec.yaml (hunk 2 above)
Comment (shan4993) [reply t/proto_template/spec.yaml:38]: ty

File: t/proto_template/.github/workflows/ci.yml (hunk 3 above)
Comment (shan4993) [reply t/proto_template/.github/workflows/ci.yml:44]: ill pass this as a variable and set 1.31 as the default

**Review by drevell — APPROVED** [review]

**Review by drevell — APPROVED** [review]

**Pull request merged**